app/models/interactive_page.rb

```diff
@@ -102,6 +102,19 @@ def main_embeddables
     section_embeddables(nil)
   end
 
+  def visible_embeddables
+    self.page_items.collect{ |qi| qi.embeddable }.select{ |e| !e.is_hidden }
```

Comment:
I was wondering if `qi` is a `page_item` so i would have expected the var to be `pi`  -- but its so obvious, it doesn't matter.  Same in the next method.